metricbeat/module/logstash/node_stats/data_xpack.go

```diff
@@ -213,26 +213,21 @@ func makeClusterToPipelinesMap(pipelines []PipelineStats, overrideClusterUUID st
 	var clusterToPipelinesMap map[string][]PipelineStats
 	clusterToPipelinesMap = make(map[string][]PipelineStats)
 
-	if overrideClusterUUID != "" {
-		clusterToPipelinesMap[overrideClusterUUID] = pipelines
-		return clusterToPipelinesMap
-	}
-
```

Comment:
I remove this section because it's also incorrect. We shouldn't be using the override cluster UUID, if set, for all pipeline so broadly. Instead we should figure out (as we do further below) the cluster UUIDs appropriate for each pipeline and build the cluster UUID => pipelines map that way.

Reply:
This is still a change in semantics and maybe a bc breaking change. 

Before this change all pipelines have had been associated with `overrideClusterUUID`. With this change pipelines are only associated with `overrideClusterUUID` iff they do no clusterUUID set. This way `overrideClusterUUID` actually becomes `defaultClusterUUID`.

Instead of removing this line, how about printing a deprecation warning if this setting is configured and introduce an a default cluster uuid setting that matches the new semantics?

Reply:
Hmmm, looking at where the value of `overrideClusterUUID` comes from ultimately, it comes from the `monitoring.cluster_uuid` setting in `logstash.yml`: https://www.elastic.co/guide/en/logstash/current/monitoring-with-metricbeat.html#define-cluster__uuid.

Reading the documentation for that setting:

> To bind the metrics of Logstash to a specific cluster, optionally define ...

This makes me think, if this setting is set and therefore `overrideClusterUUID` is set, we _should_ make that the cluster UUID for all pipelines, ignoring what Elasticsearch clusters each pipeline might individually be talking to. 

So I'm going to revert this change and, in fact, add a similar code block in the `makeClusterToPipelinesMap` function in the `node` metricset.